src/domains/chart/components/chart-with-loader.tsx

```diff
@@ -207,47 +189,66 @@ export const ChartWithLoader = ({
+        || Math.round(chartWidth / getChartPixelsPerPoint({ attributes, chartSettings }))
+      const points = forceDataPoints || dataPoints * pointsMultiplier
 
       const group = attributes.method || window.NETDATA.chartDefaults.method
 
       setShouldFetch(false)
-      dispatch(fetchDataAction.request({
-        // properties to be passed to API
-        host,
-        chart: actualChartMetadata.id,
-        format: chartSettings.format,
-        points,
-        group,
-        gtime: attributes.gtime || 0,
-        options: getChartURLOptions(attributes, shouldEliminateZeroDimensions),
-        after: after || null,
-        before: before || null,
-        dimensions: attributes.dimensions,
-
-        // properties for the reducer
-        fetchDataParams: {
-          // we store it here so it is only available when data is fetched
-          // those params should be synced with data
-          isRemotelyControlled,
-          viewRange,
-        },
-        id: chartUuid,
-        cancelTokenSource,
-      }))
+      dispatch(
+        fetchDataAction.request({
+          // properties to be passed to API
+          host,
+          chart: actualChartMetadata.id,
+          format: chartSettings.format,
+          points,
+          group,
+          gtime: attributes.gtime || 0,
+          options: getChartURLOptions(attributes, shouldEliminateZeroDimensions),
+          after: after || null,
+          before: before || null,
+          dimensions: attributes.dimensions,
+
+          // properties for the reducer
+          fetchDataParams: {
+            // we store it here so it is only available when data is fetched
+            // those params should be synced with data
+            isRemotelyControlled,
+            viewRange,
+          },
+          id: chartUuid,
+          cancelTokenSource,
+        }),
+      )
     }
-  }, [attributes, actualChartMetadata, chartSettings, chartUuid, chartWidth, dispatch,
-    hasLegend, host, initialAfter, initialBefore, isPanAndZoomMaster,
-    isRemotelyControlled, panAndZoom, portalNode, setShouldFetch, shouldEliminateZeroDimensions,
-    shouldUsePanAndZoomPadding, shouldFetch, cancelTokenSource])
+  }, [
+    attributes,
+    actualChartMetadata,
+    chartSettings,
+    chartUuid,
+    chartWidth,
+    dispatch,
+    hasLegend,
+    host,
+    initialAfter,
+    initialBefore,
+    isPanAndZoomMaster,
+    isRemotelyControlled,
+    panAndZoom,
+    portalNode,
+    setShouldFetch,
+    shouldEliminateZeroDimensions,
+    shouldUsePanAndZoomPadding,
+    shouldFetch,
+    cancelTokenSource,
+  ])
 
   const [selectedDimensions, setSelectedDimensions] = useState<string[]>([])
 
-  const hasEmptyData = (chartData as DygraphData | D3pieChartData | null)?.result.data?.length === 0
+  const hasEmptyData = (chartData as DygraphData | D3pieChartData | null)?.result?.data?.length === 0
     || (chartData as EasyPieChartData | null)?.result.length === 0
```

Comment:
```suggestion
    || (chartData as EasyPieChartData | null)?.result?.length === 0
```
if `chartData` is empty object, then this second branch will trigger an error, we need to have similar handling in the second case